components/example-notebook-servers/jupyter-pytorch-gaudi/Dockerfile

```diff
@@ -0,0 +1,143 @@
+#
+# NOTE: Use the Makefiles to build this image correctly.
+#
+
+ARG BASE_IMG=<jupyter>
+FROM $BASE_IMG
+
+# Content below is extracted from scripts/Dockerfiles here:
+# https://github.com/HabanaAI/Setup_and_Install/tree/main/dockerfiles/base
+# https://github.com/HabanaAI/Setup_and_Install/tree/main/dockerfiles/pytorch
+
+# version details
+ARG ARTIFACTORY_URL=vault.habana.ai
+ARG VERSION=1.16.2
+ARG REVISION=2
+ARG PYTORCH_VERSION=2.2.2
+ARG OS_NUMBER=2204
+
+# runtime variables
+ENV DEBIAN_FRONTEND=noninteractive
+ENV GC_KERNEL_PATH=/usr/lib/habanalabs/libtpc_kernels.so
+ENV HABANA_LOGS=/var/log/habana_logs/
+ENV HABANA_SCAL_BIN_PATH=/opt/habanalabs/engines_fw
+ENV HABANA_PLUGINS_LIB_PATH=/opt/habanalabs/habana_plugins
+ENV TCMALLOC_LARGE_ALLOC_REPORT_THRESHOLD=7516192768
+ENV PYTHONPATH=/home/$NB_UID:/usr/lib/habanalabs/
+
+# There is no need to store pip installation files inside docker image
+ENV PIP_NO_CACHE_DIR=on
+ENV PIP_DISABLE_PIP_VERSION_CHECK=1
+
+# libfabric and scaling variables
+ENV LIBFABRIC_VERSION="1.20.0"
+ENV LIBFABRIC_ROOT="/opt/habanalabs/libfabric-${LIBFABRIC_VERSION}"
+ENV MPI_ROOT=/opt/amazon/openmpi
+ENV LD_LIBRARY_PATH=$LIBFABRIC_ROOT/lib:${MPI_ROOT}/lib:/usr/lib/habanalabs:$LD_LIBRARY_PATH
+ENV PATH=${LIBFABRIC_ROOT}/bin:${MPI_ROOT}/bin:$PATH
+ENV MPICC=${MPI_ROOT}/bin/mpicc
+ENV OPAL_PREFIX=${MPI_ROOT}
+ENV RDMAV_FORK_SAFE=1
+ENV FI_EFA_USE_DEVICE_RDMA=1
+ENV RDMA_CORE_ROOT=/opt/habanalabs/rdma-core/src
+ENV RDMA_CORE_LIB=${RDMA_CORE_ROOT}/build/lib
```

Comment:
As there are a lot of envs being set here, some suggestions:

1. If they are doing things like updating the PATH or setting configs, please place them above/below the `RUN` command which is adding that package (rather than in a common section at the top of the file).
2. For the ones which are left (e.g. not related to installing a package), please group them logically underneath `#` comments which explain why they are needed and if necessary link to some external docs.

Reply:
sure, I'll move these so that they are next to the relevant RUNs